src/lib/mongoBackend/mongoSubCache.cpp

```diff
@@ -68,12 +69,21 @@ using namespace mongo;
 *
 * EntityInfo::EntityInfo - 
 */
-EntityInfo::EntityInfo(const std::string& idPattern, const std::string& _entityType)
+EntityInfo::EntityInfo(const std::string& _entityId, const std::string& _entityType, const std::string& _isPattern)
 {
-  regcomp(&entityIdPattern, idPattern.c_str(), 0);
+  entityId     = _entityId;
+  entityType   = _entityType;
+  isPattern    = (_isPattern == "true") || (_isPattern == "TRUE") || (_isPattern == "True");
 
-  entityType               = _entityType;
-  entityIdPatternToBeFreed = true;
+  if (isPattern)
+  {
+    regcomp(&entityIdPattern, _entityId.c_str(), 0);
```

Comment:
Regular expression is compiled just one time, in the moment the cache entry (in the isPattern=true case) is inserted. Is that correct?

(It seems so, but given that this is an important point, I prefer to ask to be 100% sure)